### PR TITLE
 Cover updateOrCreateFile in the crowdin.ts class #21

### PR DESCRIPTION
**Pull request by MaurizioPz**
Partially handles #21 
I did the first function in the class. Could you check if this is more or less what you where expecting and provide any feedback necessary, so that I can proceed with the rest of the class?

Thanks

### CODE REVIEW

**Comment by andrii-bodnar** [comment]
@MaurizioPz thanks a lot for your contribution!
Some [checks](https://github.com/crowdin/crowdin-apps-functions/pull/22/checks?check_run_id=9015446475) have failed, please take a look.

**Comment by andrii-bodnar** [comment]
@yevheniyJ could you please review this PR?

**Comment by MaurizioPz** [comment]
> @MaurizioPz thanks a lot for your contribution! Some [checks](https://github.com/crowdin/crowdin-apps-functions/pull/22/checks?check_run_id=9015446475) have failed, please take a look.

Yes... I was trying to have a look I at, but I don't understand what's happening, the undefined reference is nowhere in my code
image
Can you give me a pointer?
Obviously the tests pass locally without giving that weird error

Thanks

**Comment by andrii-bodnar** [comment]
@MaurizioPz it could be related to the Node version

**Comment by andrii-bodnar** [comment]
I would suggest trying to update the Node version used in CI to version 12

https://github.com/crowdin/crowdin-apps-functions/tree/main/ci

File: jestconfig.json [hunk 1]
```diff
@@ -12,5 +12,13 @@
         "jsx",
         "json",
         "node"
+    ],
+    "globals": {
+        "ts-jest": {
+          "compiler": "ttypescript"
```
Comment (andrii-bodnar) [inline jestconfig.json:18]: Is the `ttypescript` correct?

Comment (MaurizioPz) [reply jestconfig.json:18]: Yes, it's required by ts-auto-mock https://typescript-tdd.github.io/ts-auto-mock/installation#packagejson--jestconfigjs-without-the-jest-scope-1

**Comment by andrii-bodnar** [comment]
@MaurizioPz all checks have passed now, thanks again for your contribution!

For me, it looks good! I need to ask @yevheniyJ to review this PR additionally

**Review by andrii-bodnar — APPROVED** [review]

File: tests/crowdin.spec.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import { updateOrCreateFile } from '../src';
+import Crowdin, { ResponseObject, SourceFilesModel, UploadStorageModel } from '@crowdin/crowdin-api-client';
+import { createMock } from 'ts-auto-mock';
```
Comment (yevheniyJ) [inline tests/crowdin.spec.ts:3]: can you please add empty lines between imports, function/test decralations etc?

File: tests/crowdin.spec.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+describe('UpdateOrCreateFile function', () => {
+    it('should create file when source file is undefined', async () => {
+        let spyProjectId: number;
+        let spyRequest: SourceFilesModel.CreateFileRequest;
+        const client = createMock<Crowdin>({
```
Comment (yevheniyJ) [inline tests/crowdin.spec.ts:8]: I guess we can create this mock on one level above and not per each test

**Review by yevheniyJ — CHANGES_REQUESTED** [review]

File: tests/crowdin.spec.ts (hunk 3 above)
Comment (MaurizioPz) [reply tests/crowdin.spec.ts]: ok

**Review by yevheniyJ — APPROVED** [review]

**Comment by andrii-bodnar** [comment]
@yevheniyJ thank you! :)

@MaurizioPz thanks again for your contribution! I'm not closing the issue to be able to link the future pull requests with that

**Comment by MaurizioPz** [comment]
> @yevheniyJ thank you! :)
> 
> @MaurizioPz thanks again for your contribution! I'm not closing the issue to be able to link the future pull requests with that

Great I'll complete the rest of the coverage and open a new pull request
thanks

**Pull request merged**